code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -154,8 +162,59 @@
 
 	modes = list(MODE_NONE = MODE_SHUTTLE, MODE_SHUTTLE = MODE_NONE)
 
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging
+	name = "atmospheric thermal imaging goggles"
+	desc = "Goggles used by Atmospheric Technician to see the thermal energy of gasses in open areas"
+	icon_state = "trayson-atmospheric-thermal"
+	inhand_icon_state = "trayson-meson"
+	glass_colour_type = /datum/client_colour/glass_colour/gray
+
+	modes = list(MODE_NONE = MODE_ATMOS_THERMAL, MODE_ATMOS_THERMAL = MODE_NONE)
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/update_icon_state()
+	icon_state = inhand_icon_state = "trayson-[mode]"
+	return ..()
+
+
+
+/proc/atmos_thermal(mob/viewer, range = 5, duration = 10)
+	if(!ismob(viewer) || !viewer.client)
+		return
+	for(var/turf/open in view(range, viewer))
+		if(open.blocks_air)
+			continue
+		var/datum/gas_mixture/environment = open.return_air()
+		var/temp = round(environment.temperature)
```

Comment:
Would it be better to use ``return_temperature``?

Reply:
it's functionally the same no?

Reply:
Basically yes, but it would help future-proof it if return_temperature is ever changed for one reason or another.